diff --git a/lib/index.cjs b/lib/index.cjs
index 2a8b0da12dd40c1abfbd6d0c4a83b2d4ad75de84..c0da9d79ae96c186507444966d0da560e70b7bb9 100644
--- a/lib/index.cjs
+++ b/lib/index.cjs
@@ -74,6 +74,10 @@ function isPlainObject(x) {
  */
 
 function print(value) {
+  if (typeof value === 'symbol') {
+    return value.toString()
+  }
+
   return typeof value === 'string' ? JSON.stringify(value) : `${value}`;
 }
 /**
diff --git a/lib/index.mjs b/lib/index.mjs
index 83032e92ecb941d6d5e6966f9ee39cb1da792e5d..417876fe58398e9afee2ad90e3fde45ce02d0644 100644
--- a/lib/index.mjs
+++ b/lib/index.mjs
@@ -70,6 +70,10 @@ function isPlainObject(x) {
  */
 
 function print(value) {
+  if (typeof value === 'symbol') {
+    return value.toString()
+  }
+
   return typeof value === 'string' ? JSON.stringify(value) : `${value}`;
 }
 /**
diff --git a/umd/superstruct.js b/umd/superstruct.js
index ace968357a3ea65107dd966890b342fda553266e..83f1bb4ab55a71adc9a6a0373a1d9aeba354fe8e 100644
--- a/umd/superstruct.js
+++ b/umd/superstruct.js
@@ -76,6 +76,10 @@
    */
 
   function print(value) {
+    if (typeof value === 'symbol') {
+      return value.toString()
+    }
+
     return typeof value === 'string' ? JSON.stringify(value) : `${value}`;
   }
   /**
